pkg/cmd/roachtest/tests/cancel.go

```diff
@@ -66,30 +68,32 @@ func registerCancel(r registry.Registry) {
 					// (either query execution error or an error indicating the
 					// absence of expected cancellation error).
 					errCh := make(chan error, 1)
-					go func(queryNum int) {
-						runnerConn := c.Conn(ctx, t.L(), 1)
+					t.Go(func(taskCtx context.Context, l *logger.Logger) error {
+						runnerConn := c.Conn(taskCtx, l, 1)
 						defer runnerConn.Close()
 						setupQueries := []string{"USE tpch;"}
 						if !useDistsql {
 							setupQueries = append(setupQueries, "SET distsql = off;")
 						}
 						for _, setupQuery := range setupQueries {
-							t.L().Printf("executing setup query %q", setupQuery)
+							l.Printf("executing setup query %q", setupQuery)
 							if _, err := runnerConn.Exec(setupQuery); err != nil {
 								errCh <- err
 								close(sem)
-								return
+								// Errors are handled in the main goroutine.
+								return nil //nolint:returnerrcheck
```

Comment:
nit: it's not immediately obvious that returning errors in `t.Go` fails the test. I think it makes sense for that to be the behavior but you have to go a couple layers and peek at the implementation to confirm that is the case.

Reply:
Agree with Darryl's comment; we should make this explicit in the interface doc. Also, this test could technically be refactored further to explicitly return `err` instead of passing it into `errCh`, but I also understand if you want to reduce the footprint of all changes in this PR.

Reply:
I had the same thought, will need to make it clear that returning an error from a task will result in a test failure. We could alternatively provide a version of the Go/GoWithCancel methods that does not take an error return.

From the footprint perspective, I didn't want to fiddle too much with already working code. But we could definitely make some of these more ergonomic. I'll create an issue for it, since engineers tend to borrow from existing code, it would be better if the examples are improved.

Reply:
> We could alternatively provide a version of the Go/GoWithCancel methods that does not take an error return.

No bother, it's not that hard to add `return nil` :)